float64 currentX
float64 currentY
float64 currentTheta